File: chroma-league/src/main/java/net/booone/chromaleague/hud/parts/dragons/DragonBar.java
package net.booone.chromaleague.hud.parts.dragons;

import net.booone.chromaleague.state.GameStateHelper;
import net.booone.razersdk.animation.AnimatedFrame;
import net.booone.razersdk.animation.Frame;
import net.booone.razersdk.animation.LayeredFrame;

public class DragonBar extends AnimatedFrame {
    private final ElderBuffAnimation elderBuffAnimation = new ElderBuffAnimation();
    private final KilledDragonsBar killedDragonsBar = new KilledDragonsBar();

    @Override
    public Frame getFrame() {
        final LayeredFrame layeredFrame = new LayeredFrame();
        layeredFrame.addFrame(killedDragonsBar);
        if (GameStateHelper.hasElderBuff()) {
            layeredFrame.addFrame(elderBuffAnimation);
        }
        addAnimationFrame(layeredFrame);
        return super.getFrame();
    }
}

File: src/hud/parts/dragons/DragonBar.ts
import {AnimatedFrame} from '../../../razer-sdk/animation/AnimatedFrame.js';
import {Frame} from '../../../razer-sdk/animation/Frame.js';
import {LayeredFrame} from '../../../razer-sdk/animation/LayeredFrame.js';
import * as GameStateHelper from '../../../state/GameStateHelper.js';
import {ElderBuffAnimation} from './ElderBuffAnimation.js';
import {KilledDragonsBar} from './KilledDragonsBar.js';

export class DragonBar extends AnimatedFrame {
    private readonly elderBuffAnimation = new ElderBuffAnimation();
    private readonly killedDragonsBar = new KilledDragonsBar();

    override pollFrame(): Frame {
        const layeredFrame = new LayeredFrame();
        layeredFrame.addFrame(this.killedDragonsBar);
        if (GameStateHelper.hasElderBuff()) {
            layeredFrame.addFrame(this.elderBuffAnimation);
        }
        this.addAnimationFrame(layeredFrame);
        return super.pollFrame();
    }
}
